tensorflow/compiler/xla/service/gpu/tests/gpu_kernel_tiling_test.cc

```diff
@@ -602,11 +769,19 @@ TEST_F(GpuKernelTilingTest, RowReductionWithSmallDimensionNotTiled) {
       ParseAndReturnVerifiedModule(kHloString, ConfigWithoutLayoutAssignment())
           .ValueOrDie();
   CompileAndVerifyIr(std::move(hlo_module),
+#if TENSORFLOW_USE_ROCM
```

Comment:
Could we at runtime instead of an ifdef here?

Reply:
guess we could try and make this a runtime check instead of an ifdef, but what would be gain by going that route?

Making this a compile time ifdef, does make the code easier to follow.

Reply:
I'd still prefer a runtime check, if possible, as in future we might want to switch to shipping both ROCM and CUDA in the same binary.

Reply:
Thought about this some more, and I do not know whether this is feasible.

Currently we can use the stream executor to do a Runtime check to determine whether we are using ROCm or CUDA. (see example here : https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc#L322-L326 )

How do I get access to a stream executor object within this test object?

Reply:
I think I have replied using an email, but just in case it didn't make it:

There are global getters. You could either do `PlatformUtil::GetDefaultPlatform()` and try to get a stream executor from there (e.g. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/tests/while_test.cc#L1267), or check whether `se::MultiPlatformManager::PlatformWithName` returns a platform for a given name (cf. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/buffer_comparator_test.cc#L35)

Having a runtime switch has another advantage that both branches would always be compiled, and editing one would not accidentally break the compilation of another.

Reply:
@cheshire 

I have have pushed out a new commit, which makes the expected_ir switch at runtime. Please re-review.  thanks

deven